interfaces/Matlab_Toolbox_Revamp/Class/Kinetics.m

```diff
@@ -0,0 +1,590 @@
+                irange = rxns;
+            else error('stoich_p requires 1 or 3 arguments.')
+            end
+            
+            for k = krange
+                for i = irange
+                    t = calllib(ct, 'kin_productStoichCoeff', ...
+                                kin.kin_id, k-1, i-1);
+                    if t ~= 0.0
+                        temp(k, i) = t;
+                    end
+                end
+            end
+            
+            n = temp;
+        end        
+        
+        function n = stoich_net(kin, species, rxns)
+            % Get the net stoichiometric coefficients. 
+            %
+            % parameter species:
+            %    Species indices for which net stoichiometric coefficients
+            %    should be retrieved. Optional argument; if specified,
+            %    "rxns" must be specified as well. 
+            % parameter rxns:
+            %    Reaction indicies for which net stoichiometric
+            %    coefficients should be retrieved. Optional argument; if
+            %    specified, "species" must be specified as well.
+            % return:
+            %    Returns a sparse matrix of all net stoichiometric
+            %    coefficients. The matrix elements "nu(k, i)" is the
+            %    stoichiometric coefficient of species k as a net in
+            %    reaction i. If "species" and "rxns" are specified, the
+            %    matrix will contain only entries for the specified species
+            %    and reactions. For example, "stoich_net(a, 3, [1, 3, 5,
+            %    7])" returns a sparse matrix containing only the
+            %    coefficients for specis 3 in reactions 1, 3, 5, and 7. 
+            
+            checklib;
+            if nargin == 1
+                n = stoich_p(kin)-stoich_r(kin);
+            elseif nargin == 3
+                n = stoich_p(a, species, rxns) - stoich_r (a, species, rxns);
+            else error('stoich_net requires 1 or 3 arguments.');
+            end
+        end
+        
+        %% Get reaction array attributes
+                
+        function cdot = creationRates(kin)
+            % Get the chemical reaction rates.
+            %
+            % return:
+            %    Returns a vector of the creation rates of all species. If
+            %    the output is not assigned to a variable, a bar graph is
+            %    produced. Unit: kmol/m^3-s.
+            
+            checklib;
+            nsp = kin.nSpecies;
+            xx = zeros(1, nsp);
+            pt = libpointer('doublePtr', xx);
+            calllib(ct, 'kin_getCreationRates', kin.kin_id, nsp, pt);
+            cdot = pt.Value;
+            if nargout == 0
+                figure
+                set(gcf, 'Name', 'Creation Rates')
+                bar(q)
+                xlabel('Species Number')
+                ylabel('Creation Rate [kmol/m^3-s]')
+                title('Species Chemical Reaction Rates')
+            end            

```

Comment:
While I am aware that the feature of 'produce a plot if no output is defined' is motivated by the behavior of the old MATLAB toolbox, I'm not sure whether it deserves to be included here: it's a lot of code that is very likely barely used. (Here and elsewhere.)